src/common/utxobased/engine/types.ts

```diff
@@ -1,7 +1,20 @@
+import { asArray, asBoolean, asObject, asOptional, asString } from 'cleaners'
 import { EdgeSpendInfo } from 'edge-core-js/types'
 
 import { Input, Output } from '../keymanager/utxopicker/types'
 
+// Normalizes a URL to always be a websocket URL
+const asWebsocketUrl = (raw: unknown): string => {
+  const url = new URL(asString(raw))
+  return `wss://${url.host}/websocket`
+}
+
+export const asUtxoUserSettings = asObject({
+  blockbookServers: asOptional(asArray(asWebsocketUrl), []),
+  enableCustomServers: asOptional(asBoolean, false)
+})
+export type UtxoUserSettings = ReturnType<typeof asUtxoUserSettings>
```

Comment:
Mirroring the concern above, should these be `asMaybe` instead of `asOptional`, so it's fool-proof?